WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigActivity.kt

```diff
@@ -1,17 +1,137 @@
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_watch_face_config)
         Log.d(TAG, "onCreate()")
+
+        binding = ActivityWatchFaceConfigBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Disable widgets until data loads and values are set.
+        binding.colorStylePickerButton.isEnabled = false
+        binding.ticksEnabledSwitch.isEnabled = false
+        binding.minuteHandLengthSlider.isEnabled = false
+
+        // Set max and min.
+        binding.minuteHandLengthSlider.valueTo = MINUTE_HAND_LENGTH_MAXIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.valueFrom = MINUTE_HAND_LENGTH_MINIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.value = MINUTE_HAND_LENGTH_DEFAULT_FOR_SLIDER
+
+        binding.minuteHandLengthSlider.addOnChangeListener { slider, value, fromUser ->
+            Log.d(TAG, "addOnChangeListener(): $slider, $value, $fromUser")
+            stateHolder.setMinuteHandArmLength(value)
+        }
+
+        lifecycleScope.launch(Dispatchers.Main.immediate) {
+            stateHolder.uiState.collect { uiState: WatchFaceConfigStateHolder.EditWatchFaceUiState ->
+                when (uiState) {
+                    is WatchFaceConfigStateHolder.EditWatchFaceUiState.Loading -> {
+                        Log.d(TAG, "StateFlow Loading: ${uiState.message}")
+                    }
+                    is WatchFaceConfigStateHolder.EditWatchFaceUiState.Success -> {
+                        Log.d(TAG, "StateFlow Success.")
+                        updateWatchFacePreview(uiState.userStylesAndPreview)
+                    }
+                    is WatchFaceConfigStateHolder.EditWatchFaceUiState.Error -> {
+                        Log.e(TAG, "Flow error: ${uiState.exception}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun updateWatchFacePreview(
+        userStylesAndPreview: WatchFaceConfigStateHolder.UserStylesAndPreview
+    ) {
+        Log.d(TAG, "updateWatchFacePreview: $userStylesAndPreview")
+
+        val colorStyleId: String = userStylesAndPreview.colorStyleId
+        Log.d(TAG, "\tselected color style: $colorStyleId")
+
+        binding.ticksEnabledSwitch.isChecked = userStylesAndPreview.ticksEnabled
+        binding.minuteHandLengthSlider.value = userStylesAndPreview.minuteHandLength
+        binding.preview.watchFaceBackground.setImageBitmap(userStylesAndPreview.previewImage)
+
+        if (!initialDataLoaded) {
+            initialDataLoaded = true
+            enabledWidgets()
+        }
+    }
+
+    private fun enabledWidgets() {
+        binding.colorStylePickerButton.isEnabled = true
+        binding.ticksEnabledSwitch.isEnabled = true
+        binding.minuteHandLengthSlider.isEnabled = true
+    }
+
+    override fun onDestroy() {
+        Log.d(TAG, "onDestroy()")
+        stateHolder.onCleared()
+        // Makes sure the activity closes.
+        finish()
```

Comment:
Just curious, why do we need this here?

Reply:
Good point, I need to add a comment.

It's called out specifically to do in the [jetpack code](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:wear/wear-watchface-editor/samples/src/main/java/androidx/wear/watchface/editor/sample/WatchFaceConfigActivity.kt;l=216?q=EditorSession&ss=androidx%2Fplatform%2Fframeworks%2Fsupport). I will start a thread with Alex and verify why... then add a follow up CL with a comment explaining why.